### PR TITLE
Enable algorithms unit tests for HIP

### PR DESCRIPTION
**Pull request by masterleinad**
With the current `concurrency` value we never get out of the `while` loop in `Random_UniqueIndex::get_state_idx`. The new value also looks closer to what was in the comments for `fiji` and `hawaii`.
@crtrott Do you recall why we chose that value as it currently is?

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Comment by crtrott** [comment]
Technically you should figure out the concurrency value from some runtime checking the GPU. We chose that value in ROCm because it was enough for any viable AMD GPU at the time. It needs to be at least as large as the maximum number of concurrently active threads on the GPU.

File: core/src/HIP/Kokkos_HIP_Space.cpp [hunk 1]
```diff
@@ -575,7 +575,8 @@ namespace Experimental {
 
 int HIP::concurrency() {
   // FIXME_HIP
-  return 32 * 8 * 40;  // 81920 fiji and hawaii
+  // MI60: ThreadsPerComputeUnit*ComputeUnits/ShaderEngine*ShaderEngines)
+  return 16 * 4 * 1024;
```
Comment (crtrott) [inline core/src/HIP/Kokkos_HIP_Space.cpp:579]: Is this right? Don't they have like 2536 threads or so per compute unit? (1024 is the limit per block or group, but you can have like 5 groups of 512 simultaneously or?)

**Comment by masterleinad** [comment]
Unfortunately, the runtime API doesn't provide these values, yet so we still have to choose something ourselves. I am/was confused by the old value because `32 * 8 * 40 = 10240` and not `81920`.

File: core/src/HIP/Kokkos_HIP_Space.cpp (hunk 1 above)
Comment (crtrott) [reply core/src/HIP/Kokkos_HIP_Space.cpp:579]: https://www.olcf.ornl.gov/wp-content/uploads/2019/10/ORNL_Application_Readiness_Workshop-AMD_GPU_Basics.pdf
Yeah its 40 wave fronts with 64 threads each per CU i.e. 2536 threads per CU.

**Review by crtrott — CHANGES_REQUESTED** [review]
The concurrency is still too low.

**Comment by masterleinad** [comment]
Thanks for correcting, @crtrott! Fixed now.

**Review by crtrott — APPROVED** [review]
Ok this is fine for now. We should alter fix here and in CUDA the concurrency to ask the runtimes for correct numbers.

**Pull request merged**